src/app/provider/view.nim

```diff
@@ -17,8 +22,85 @@ QtObject:
     result.status = status
     result.setup
 
-  proc postMessage*(self: Web3ProviderView, data: string): string {.slot.} =
-    # TODO: implement code from status-react/src/status_im/browser/core.cljs
-    echo "==========================================="
-    echo "Message received from JS web3 provider: ", data
-    return "Hello World!" # This can only be seen in chrome devtools
\ No newline at end of file
+  proc web3AsyncReadOnly*(self: Web3ProviderView, data: JsonNode): JsonNode =
+    let messageId = data["messageId"]
+    let messageType = "web3-send-async-callback"
+    let payloadId = data["payload"]["id"]
+    let rpcMethod = data["payload"]["method"].getStr()
+
+    let authMethods = toHashSet(["eth_accounts", "eth_coinbase", "eth_sendTransaction", "eth_sign", "keycard_signTypedData", "eth_signTypedData", "personal_sign", "personal_ecRecover"])
+    let signMethods = toHashSet(["eth_sendTransaction", "personal_sign", "eth_signTypedData", "eth_signTypedData_v3"])
+    let accMethods = toHashSet(["eth_accounts", "eth_coinbase"])
```

Comment:
Probably a good idea to pull this out in to a `const`, as it doesn't need to be done with every request.